protocol/app/ante/market_update_test.go

```diff
@@ -0,0 +1,538 @@
+							sdk.NewCoin(assets.AssetUsdc.Denom, sdkmath.NewInt(1)),
+						},
+					},
+					&banktypes.MsgSend{
+						FromAddress: constants.BobAccAddress.String(),
+						ToAddress:   constants.AliceAccAddress.String(),
+						Amount: []sdk.Coin{
+							sdk.NewCoin(assets.AssetUsdc.Denom, sdkmath.NewInt(1)),
+						},
+					},
+				},
+				simulate: true,
+			},
+			wantErr: false,
+		},
+		{
+			name: "do nothing for non-market map messages",
+			args: args{
+				msgs: []sdk.Msg{
+					&banktypes.MsgSend{
+						FromAddress: constants.BobAccAddress.String(),
+						ToAddress:   constants.AliceAccAddress.String(),
+						Amount: []sdk.Coin{
+							sdk.NewCoin(assets.AssetUsdc.Denom, sdkmath.NewInt(1)),
+						},
+					},
+					&banktypes.MsgSend{
+						FromAddress: constants.BobAccAddress.String(),
+						ToAddress:   constants.AliceAccAddress.String(),
+						Amount: []sdk.Coin{
+							sdk.NewCoin(assets.AssetUsdc.Denom, sdkmath.NewInt(1)),
+						},
+					},
+				},
+				simulate: false,
+			},
+			wantErr: false,
+		},
+		{
+			name: "reject for multiple market messages - simulate",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+				},
+				simulate: true,
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject for multiple market messages",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+				},
+				simulate: false,
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject for multiple mixed market messages - simulate",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+					&mmtypes.MsgUpdateMarkets{
+						Authority:     constants.BobAccAddress.String(),
+						UpdateMarkets: nil,
+					},
+				},
+				simulate: true,
+			},
+			wantErr: true,
+		},
+		{
+			name: "reject for multiple mixed market messages",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+					&mmtypes.MsgUpdateMarkets{
+						Authority:     constants.BobAccAddress.String(),
+						UpdateMarkets: nil,
+					},
+				},
+				simulate: false,
+			},
+			wantErr: true,
+		},
+		{
+			name: "accept a single message with no markets",
+			args: args{
+				msgs: []sdk.Msg{
+					&mmtypes.MsgUpsertMarkets{
+						Authority: constants.BobAccAddress.String(),
+						Markets:   nil,
+					},
+				},
+				simulate: false,
+			},
+			wantErr: false,
+		},
+		{
+			name: "accept a single message with no cross markets (only isolated)",
```

Comment:
nit: Test-case where no perps have a market id that matches the market being updated?